packages/@lwc/engine/src/framework/base-lightning-element.ts

```diff
@@ -287,6 +288,7 @@ function BaseLightningElementConstructor(this: LightningElement) {
         mode,
         delegatesFocus: !!ctor.delegatesFocus,
     };
+    assign(shadowRootOptions, { '$$lwc-synthetic-mode$$': true });
```

Comment:
Instead of creating the `shadowRootOptions` init and then assigning the `$$lwc-synthetic-mode$$` property, I would recommend for performance to move the `$$lwc-synthetic-mode$$` in the object literal.